src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,191 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head = null;
+    private Node<T> tail = null;
+    private int size = 0;
+
+    static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            Node<T> newNode = new Node<T>(null, value, null);
+            head = newNode;
+            tail = head;
+        } else {
+            Node<T> newNode = new Node<T>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> currentNode;
+        if (index == 0) {
+            if (head == null) {
+                Node<T> newNode = new Node<T>(null, value, null);
+                head = newNode;
+                tail = head;
```

Comment:
duplicates code from add(T value) method
```suggestion
                add(value);
                return;
```